docling_jobkit/orchestrators/rq/orchestrator.py

```diff
@@ -0,0 +1,146 @@
+        self,
+        config: RQOrchestratorConfig,
+        converter_manager_config: DoclingConverterManagerConfig,
+        api_only=False,
+    ):
+        super().__init__()
+        self.config = config
+        self.api_only = api_only
+        self.worker_processes: list[Popen] = []
+        self.redis_conn = Redis(
+            host=os.environ.get("DOCLING_SERVE_eng_rq_host", "localhost"),
+            port=os.environ.get("DOCLING_SERVE_eng_rq_port", 6379),
+        )
+        self.task_queue = Queue(
+            "conversion_queue", connection=self.redis_conn, default_timeout=14400
+        )
+        self.cm_conf = converter_manager_config
+
+    async def notify_end_job(self, task_id):
+        # TODO: check if this is necessary
+        pass
+
+    async def enqueue(
+        self,
+        sources: list[TaskSource],
+        options: ConvertDocumentsOptions,
+        target: TaskTarget,
+    ) -> Task:
+        task_id = str(uuid.uuid4())
+        task = Task(task_id=task_id, sources=sources, options=options, target=target)
+        self.tasks.update({task.task_id: task})
+        task_data = task.model_dump(mode="json")
+        self.task_queue.enqueue(
+            conversion_task,
+            kwargs={"task_data": task_data},
+            job_id=task_id,
+            timeout=14400,
+        )
+        await self.init_task_tracking(task)
+
+        return task
+
+    async def queue_size(self) -> int:
+        return self.task_queue.count
+
+    async def get_queue_position(self, task_id: str) -> Optional[int]:
+        try:
+            # On fetching Job to get queue position, we also get the status
+            # in order to keep the status updated in the tasks list
+            job = Job.fetch(task_id, connection=self.redis_conn)
+            status = job.get_status()
+            queue_pos = job.get_position()
+            if status == JobStatus.FINISHED:
+                task = self.tasks[task_id]
+                task.task_status = TaskStatus.SUCCESS
+                task.results = job.return_value()
+                self.tasks.update({task.task_id: task})
+            elif status == JobStatus.QUEUED or status == JobStatus.SCHEDULED:
+                task = self.tasks[task_id]
+                task.task_status = TaskStatus.PENDING
+                self.tasks.update({task.task_id: task})
+            elif status == JobStatus.STARTED:
+                task = self.tasks[task_id]
+                task.task_status = TaskStatus.STARTED
+                self.tasks.update({task.task_id: task})
+            else:
+                task = self.tasks[task_id]
+                task.task_status = TaskStatus.FAILURE
+                self.tasks.update({task.task_id: task})
+            return queue_pos + 1 if queue_pos is not None else 0
+        except Exception as e:
+            _log.error("An error occour getting queue position.", exc_info=e)
+            return None
+
+    async def process_queue(self):
+        if not self.api_only:
```

Comment:
is this only for local debugging? I don't think we want to launch the subprocesses in this way.